pkg/bindings/connection.go

```diff
@@ -108,23 +111,76 @@ func NewConnectionWithIdentity(ctx context.Context, uri string, identity string,
 	var connection Connection
 	switch _url.Scheme {
 	case "ssh":
+		userinfo := _url.User
```

Comment:
Can you wrap the entire part in a new function please like tcpClient()

Reply:
Added here: https://github.com/containers/podman/pull/23893